python/lsst/cp/pipe/cpCombine.py

```diff
@@ -612,6 +615,33 @@ def setFilter(exp, filterLabel):
         """
         pass
 
+    def calibStats(self, exp, calibrationType):
+        """Measure bulk statistics for the calibration.
+
+        Parameters
+        ----------
+        exp : `lsst.afw.image.Exposure`
+            Exposure to calculate statistics for.
+        calibrationType : `str`
+            Type of calibration to record in header.
+        """
+        metadata = exp.getMetadata()
+
+        # percentiles
+
+        for amp in exp.getDetector():
+            ampImage = exp[amp.getBBox()]
+            p0, p5, p16, p50, p84, p95, p100 = np.percentile(ampImage.image.array,
+                                                             [0, 5, 16, 50, 84, 95, 100])
+
+            metadata.set(f"LSST CALIB {calibrationType.upper()} {amp.getName()} DISTRIBUTION 0-PCT", p0)
+            metadata.set(f"LSST CALIB {calibrationType.upper()} {amp.getName()} DISTRIBUTION 5-PCT", p5)
+            metadata.set(f"LSST CALIB {calibrationType.upper()} {amp.getName()} DISTRIBUTION 16-PCT", p16)
+            metadata.set(f"LSST CALIB {calibrationType.upper()} {amp.getName()} DISTRIBUTION 50-PCT", p50)
+            metadata.set(f"LSST CALIB {calibrationType.upper()} {amp.getName()} DISTRIBUTION 84-PCT", p84)
+            metadata.set(f"LSST CALIB {calibrationType.upper()} {amp.getName()} DISTRIBUTION 95-PCT", p95)
+            metadata.set(f"LSST CALIB {calibrationType.upper()} {amp.getName()} DISTRIBUTION 100-PCT", p100)
+
```

Comment:
Instead of hard-coding these, could we take the percentile values to compute as an input, and loop over those?

Reply:
You're right, hard coding this is likely to be annoying in the future.  I've moved the percentile levels to a config option, and have this method `zip` the levels and measured values into a second loop.  Then, if someone decides the quartiles are useful too, it's just a config change.

Reply:
PS `metadata[key] = value` should work fine.